ddtrace/span.py

```diff
@@ -277,6 +280,13 @@ def _finish_ns(self, finish_time_ns):
         for cb in self._on_finish_callbacks:
             cb(self)
 
+    def _override_sampling_decision(self, decision):
+        self.context.sampling_priority = decision
+        set_sampling_decision_maker(self.context, SamplingMechanism.MANUAL)
+        for key in (SAMPLING_RULE_DECISION, SAMPLING_AGENT_DECISION, SAMPLING_LIMIT_DECISION):
```

Comment:
we make sure to delete all of the tags that shouldn't be on the span if we make a manual keep or drop decision.